### PR TITLE
TYP: resolve ignored mypy errors in core/describe.py

### PR DESCRIPTION
**Pull request by iasoon**
xref https://github.com/pandas-dev/pandas/issues/37715

### CODE REVIEW

File: pandas/io/formats/format.py [hunk 1]
```diff
@@ -1667,7 +1667,7 @@ def _format_strings(self) -> list[str]:
 
 
 def format_percentiles(
-    percentiles: (np.ndarray | list[int | float] | list[float] | list[str | float]),
+    percentiles: (np.ndarray | Sequence[int | float]),
```
Comment (twoertwein) [inline pandas/io/formats/format.py:1670]: I think you can use just `Sequence[float]` (I think type checkers wiredly treat int as a sub-class of float). If you need the union, please use `Sequence[float] | Sequence[int]`.

**Review by twoertwein — APPROVED** [review]
Small comment, looks good otherwise!

File: pandas/io/formats/format.py (hunk 1 above)
Comment (iasoon) [reply pandas/io/formats/format.py]: That seems to be correct: https://mypy.readthedocs.io/en/latest/duck_type_compatibility.html
I adapted the code.

File: pandas/io/formats/format.py [hunk 2]
```diff
@@ -1667,7 +1667,7 @@ def _format_strings(self) -> list[str]:
 
 
 def format_percentiles(
-    percentiles: (np.ndarray | list[int | float] | list[float] | list[str | float]),
+    percentiles: (np.ndarray | Sequence[float]),
```
Comment (twoertwein) [inline pandas/io/formats/format.py:1670]: Was there a reason why the function was supposed to accept strings? @jreback I don't think it supports str as one of the first lines is`np.isclose(percentiles.astype(int), percentiles)` which should fail for str (didn't try it)?

Comment (jreback) [reply pandas/io/formats/format.py:1670]: don't remember. happy to eliminate it (assume we don't have tests / usage for this)

**Comment by jreback** [comment]
thanks @iasoon

**Pull request merged**